src/file_pipe_log/reader.rs

```diff
@@ -64,17 +65,49 @@ impl<F: FileSystem> LogItemBatchFileReader<F> {
     /// Returns the next [`LogItemBatch`] in current opened file. Returns
     /// `None` if there is no more data or no opened file.
     pub fn next(&mut self) -> Result<Option<LogItemBatch>> {
-        if self.valid_offset < self.size {
+        // @lucasliang.
+        // TODO: [Fulfilled in writing progress when DIO is open.]
+        // We should also consider that there might exists broken blocks when DIO
+        // is open, and the following reading strategy should tolerate reading broken
+        // blocks until it finds an accessible header of `LogBatch`.
+        let fs_block_size = get_system_block_size();
+        while self.valid_offset < self.size {
+            debug_assert!(self.file_context.is_some());
+            let data_layout = self.file_context.as_ref().unwrap().format.data_layout();
             if self.valid_offset < LOG_BATCH_HEADER_LEN {
                 return Err(Error::Corruption(
                     "attempt to read file with broken header".to_owned(),
                 ));
             }
-            let (footer_offset, compression_type, len) = LogBatch::decode_header(&mut self.peek(
-                self.valid_offset,
-                LOG_BATCH_HEADER_LEN,
-                0,
-            )?)?;
+            let (footer_offset, compression_type, len) = {
```

Comment:
This code is ugly, how about separate the two implementations clearly like this:
```
match data_layout {
  DataLayout::NoAlignment => { old code }
  DataLayout::Alignment(size) => {
    new code
  }
}
```

Reply:
Both accepted, current format is used to seperate the extraction of `(footer_offset, compression_type, len)` and share the following decoding strategy of LogBatch.

Reply:
I perfer current one? Because the strategy for loading header of LogBatch are diverged when `DataLayout` are different, and the following processing on reading and parsing LogBatch content could be kept in same code paths.

Reply:
Or how about sperate the decoding of header into divided parts, such as:
``` Rust
let (footer_offset, compression_type, len) = {
       match data_layout {
            DataLayout::NoAlignment => {
                ... /* get tuple of (offset, t, l) */
            }
            DataLayout::Alignment(fs_block_size) => {
                ... /* get tuple of (offset, t, l) */
            }
        }
    };
/* Old codes for decoding LogBatch. */
```